### DISCUSSION TITLE
Alpine only partially initiates

### DISCUSSION THREAD

**Discussion by millsp** [alpinejs/alpine#4028] [7. Bugs (archived)]
This simple example does not work, and no alert is shown. However, by adding `x-init` somehow makes it initialize properly. While this example and workaround are very simple, it is worth adding that it seems this problem "branches out": in other words, as long as `x-on:click` is not surrounded by an `x-data` or `x-init` it will not trigger, although Alpine has been loaded correctly.
```html
<html lang="en-US">
  <head>
    <title>Home</title>
    <script defer="" src="https://cdn.jsdelivr.net/npm/alpinejs@3.13.5/dist/cdn.min.js"></script>
  </head>
  <body>
    <button x-on:click="alert('Hello World!')">Say Hi</button>
  </body>
</html>
```

```html
<html lang="en-US">
  <head>
    <title>Home</title>
    <script defer="" src="https://cdn.jsdelivr.net/npm/alpinejs@3.13.5/dist/cdn.min.js"></script>
  </head>
  <body x-init="{}">
    <button x-on:click="alert('Hello World!')">Say Hi</button>
  </body>
</html>
```

**Comment by SimoTod** [comment]
You need x-data to define an alpine component, otherwise none of the directives will work (exept for x-init). 

It's been raised a few times (if you search, you'll see other duscussions).
The reason why the examples don't have x-data is because people will just copy and paste without understanding, ending up with an x-data directive for each html tag.
